### PR TITLE
Updates to post_type function

### PR DESCRIPTION
**Pull request by patphg**
Allows for custom icon classes and the ability to toggle linking to the post type archive. Text is now populated by labels defined in the post type to accommodate for custom text.

In my testing this is not a breaking change to existing calls to the function.

### CODE REVIEW

File: libs/Functions.php [hunk 1]
```diff
@@ -511,16 +511,24 @@ public function post_author($options=array()){
...
 		if ( get_theme_mod('search_result_post_types', true) ){
-			$post_icon_img = '<i class="fas fa-thumbtack"></i>';
-			if ( $icon ){
+			if( $data['icon'] ){
 				global $wp_post_types;
 				$post_type = get_post_type();
+				$post_type_labels = get_post_type_object( $post_type )->labels;
 
-				if ( $post_type === 'post' ){
+				if( gettype( $data['icon'] ) == 'string' && $data['icon'] != '' ){
```
Comment (chrisblakley) [inline libs/Functions.php:527]: Please use `===` and `!==`

File: libs/Functions.php [hunk 2]
```diff
@@ -534,10 +542,16 @@ public function post_type($icon=true){
 						$post_icon_img = '<i class="fas fa-thumbtack"></i>';
 					}
 				}
+			}else{
+				$post_icon_img = false;
```
Comment (chrisblakley) [inline libs/Functions.php:546]: Should be able to set `$post_icon_img = false;` first before the `if` which would avoid the need for an `else` altogether.

File: libs/Functions.php [hunk 3]
```diff
@@ -534,10 +542,16 @@ public function post_type($icon=true){
...
 			}
 
-			return '<span class="meta-item post-type">' . $post_icon_img . ucwords(get_post_type()) . '</span>';
-		}
+			if( $data['linked'] ){
+				return '<span class="meta-item post-type"><a href="'.esc_url( get_post_type_archive_link( $post_type ) ).'" title="See all '.$post_type_labels->name.'">'.$post_icon_img.esc_html($post_type_labels->singular_name).'</a></span>';
+			}else{
```
Comment (chrisblakley) [inline libs/Functions.php:551]: `else` is not necessary here– return within the first `if` and then simply return the second time without the need for `else`.

File: libs/Functions.php [hunk 4]
```diff
@@ -511,16 +511,24 @@ public function post_author($options=array()){
...
-			$post_icon_img = '<i class="fas fa-thumbtack"></i>';
-			if ( $icon ){
+			if( $data['icon'] ){
 				global $wp_post_types;
 				$post_type = get_post_type();
+				$post_type_labels = get_post_type_object( $post_type )->labels;
 
-				if ( $post_type === 'post' ){
+				if( gettype( $data['icon'] ) == 'string' && $data['icon'] != '' ){
+					$post_icon_img = '<i class="'.esc_html( $data['icon'] ).'"></i>';
```
Comment (chrisblakley) [inline libs/Functions.php:528]: I wrap concatenation `.` with spaces– not a huge problem as I can go through and modify that for consistency myself, but noting it here and below as well.

File: libs/Functions.php [hunk 5]
```diff
@@ -534,10 +542,16 @@ public function post_type($icon=true){
...
 					}
 				}
+			}else{
+				$post_icon_img = false;
 			}
 
-			return '<span class="meta-item post-type">' . $post_icon_img . ucwords(get_post_type()) . '</span>';
-		}
+			if( $data['linked'] ){
+				return '<span class="meta-item post-type"><a href="'.esc_url( get_post_type_archive_link( $post_type ) ).'" title="See all '.$post_type_labels->name.'">'.$post_icon_img.esc_html($post_type_labels->singular_name).'</a></span>';
```
Comment (chrisblakley) [inline libs/Functions.php:550]: Same note about space around `.` here too. Just noting for sake of review completeness.

File: libs/Functions.php [hunk 6]
```diff
@@ -534,10 +542,16 @@ public function post_type($icon=true){
...
+			}else{
+				return '<span class="meta-item post-type">'.$post_icon_img.esc_html($post_type_labels->singular_name).'</span>';
```
Comment (chrisblakley) [inline libs/Functions.php:552]: Same note about space around `.` here too. Just noting for sake of review completeness.

File: libs/Functions.php [hunk 7]
```diff
@@ -511,16 +511,24 @@ public function post_author($options=array()){
...
 
-				if ( $post_type === 'post' ){
+				if( gettype( $data['icon'] ) == 'string' && $data['icon'] != '' ){
+					$post_icon_img = '<i class="'.esc_html( $data['icon'] ).'"></i>';
+				}elseif ( $post_type === 'post' ){
 					$post_icon_img = '<i class="fas fa-fw fa-thumbtack"></i>';
-				} elseif ( $post_type === 'page' ){
+				} elseif ( $post_type === 'page' ){		
 					$post_icon_img = '<i class="fas fa-fw fa-file-alt"></i>';
 				} else {
```
Comment (chrisblakley) [inline libs/Functions.php]: Technically this `else` is also unnecessary– I know that is carried over from the original writing, so I may re-visit that part myself later.

File: libs/Functions.php [hunk 8]
```diff
@@ -511,16 +511,24 @@ public function post_author($options=array()){
 	}
 
 	//Post type meta
-	public function post_type($icon=true){
+	public function post_type($options=array() ){
+		$defaults = apply_filters('nebula_post_type_defaults', array(
+			'icon' 		=> true, // boolean true for generic defaults, boolean false to disable icon, or string of class name(s) for icon.
```
Comment (chrisblakley) [inline libs/Functions.php:516]: For consistency with the rest of Nebula, we can remove the tabs for the array definitions.

**Review by chrisblakley — CHANGES_REQUESTED** [review]
This looks really good– I noted some minor changes that we can either adjust before or even after merging this PR. I love the idea of using the actual singular/plural label– great thought.

File: libs/Functions.php (hunk 7 above)
Comment (patphg) [reply libs/Functions.php:534]: Only you can conquer your past mistakes

Comment (patphg) [reply libs/Functions.php:534]: I refactored this while fixing another issue anyway

**Pull request merged**